### ISSUE TITLE
Input Object fields

### ISSUE THREAD

**Issue by HurricaneJames** [facebook/graphql#85]
Should input objects return a keyed set when introspecting for `inputFields` the way object do for `fields`?

Currently, the spec on introspection says that input objects have an `inputFields` query that returns a list. Object, on the other hand, have a `fields` query that returns a keyed set. Both input objects and objects take maps, even if the keys are just 0, 1, 2, 3. Is there a reason that introspection does not return a keyed set of fields for input objects? In a similar vein, why not just have the introspection query be 'fields' for both?

**Comment by leebyron** [comment]
Can you share the introspection query in particular? GraphQL does not have a concept of keyed sets or maps, only lists, so I'm a little confused with the issue.

inputFields and fields have different names in introspection because they return different types. Since introspection uses one simplified and flattened "__Type" type that can represent all of the types in a type system, it needs differently named fields to refer to these two different concepts.

**Comment by HurricaneJames** [comment]
Apologies, I think I was confused. No, I was definitely confused. There is not a lot of literature on mutations in GraphQL, and therefore not a lot of literature on really big blobs of data in Input Objects. I was trying to tease out the implementation of GraphQL-JS vs the actual spec and Object vs Input Object. 

I went back and ran the introspections again now that I have significantly more experience with Input Object, I think perhaps I was leaving out the name field, which made the two seem different. Also, defining `fields` on the GraphQL-JS GraphQLInputObject, but introspecting on `inputFields` was causing some confusion (I created a separate ticket for that issue on that project (graphql/graphql-js#144).

Sorry for wasting your time on this.
